tangelo/toolboxes/operators/trim_trivial_qubits.py

```diff
@@ -0,0 +1,143 @@
+        trim_index (list):  index of qubits to trim
+        trim_states (list): state of the qubits to trim, 0 or 1
+        reindex (bool): If True, remaining qubits will be reindexed
+    Returns:
+        qu_op  (QubitOperator): trimmed QubitOperator with updated coefficients
+    """
+    trim_states = [x for (y, x) in sorted(zip(trim_index, trim_states), key=lambda pair: pair[0])]
+    trim_index = sorted(trim_index)
+    qu_op_trim = QubitOperator()
+
+    # Calculate expectation values of trivial qubits, update coefficients
+    for op, coeff in qu_op.terms.items():
+        term = pauli_of_to_string(op, n_qubits)
+        c = np.ones(len(trim_index))
+        new_term = term
+        for i, qubit in enumerate(trim_index):
+            if term[qubit] in ('X', 'Y'):
+                c[i] = 0
+            elif term[qubit] == 'Z' and trim_states[i] == 1:
+                c[i] = -1
+
+            new_term = new_term[:qubit - i] + new_term[qubit - i + 1:] if reindex else new_term[:qubit] + 'I' + new_term[qubit + 1:]
+
+        qu_op_trim += np.prod(c) * coeff * QubitOperator(pauli_string_to_of(new_term))
+    return qu_op_trim
+
+
+def trim_trivial_circuit(circuit):
+    """
+        Splits Circuit into entangled. and unentangled components.
+        Returns entangled Circuit, and the indices and states of unentangled qubits
+
+        Args:
+            circuit (Circuit): circuit to be trimmed
+        Returns:
+            circuit_new (Circuit) : Trimmed, entangled circuit
+            trim_states (list): state of removed qubits, 0 or 1
+            trim_index (list):  index of removed qubits
+
+    """
+    # Split circuit into components with entangling and nonentangling gates
+    circs = circuit.split()
+    e_indices = circuit.get_entangled_indices()
+
+    # Get list of qubits with gates applied to them
+    gated_qubits = [qubit for e_subset in e_indices for qubit in e_subset]
+
+    # Find qubits with no gates applied to them, store qubit index and state |0>
+    zeros = list(sorted(set(range(circuit.width)) - set(gated_qubits)))
+    trim_index = [i for i in zeros]
+    trim_states = [0 for i in range(len(zeros))]
+
+    circuit_new = Circuit()
+    # Go through circuit components, trim if trivial, otherwise append to new circuit
+    for i, circ in enumerate(circs):
+        if circ.width != 1:
+            circuit_new += circ
+            continue
+        # Calculate state of single qubit clifford circuits, ideally this would be done with a clifford simulator
+        # for now only look at first two gate combinations typical of the QMF state in QCC methods
+        num_gates = len(circ._gates)
+        if num_gates not in (1, 2):
+            circuit_new += circ
+            continue
+
+        gate0, gate1 = circ._gates[:2] + [None] * (2 - num_gates)
+        gate_0_is_clifford = True if gate0 is not None and (gate0.parameter == "" or np.isclose(gate0.parameter, np.pi)) else False
```

Comment:
If this applies to any multiple of pi, I'd suggest upgrading to
`np.isclose(gate0.parameter%np.pi, 0)`

Also, your whole `if` statement is already a True / False expression ! You can just return it directly here:

`gate_0_is_clifford = (gate0 is not None and (gate0.parameter == "" or np.isclose(...)))`